### PR TITLE
fix(upgrade): add invalid project condition

### PR DESCRIPTION
**Pull request by xzf0587**
fix bug: https://dev.azure.com/msazure/Microsoft%20Teams%20Extensibility/_workitems/edit/17900027

### CODE REVIEW

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 1]
```diff
@@ -179,11 +179,11 @@ export const ProjectMigratorMWV3: Middleware = async (ctx: CoreHookContext, next
...
-  } else if (versionForMigration.state === VersionState.upgradeable && checkMethod(ctx)) {
-    if (!checkUserTasks(ctx)) {
-      ctx.result = ok(undefined);
-      return;
-    }
+  }
+  const projectPath = getParameterFromCxt(ctx, "projectPath", "");
+  const isValid = await checkActiveResourcePlugins(projectPath);
+  const isUpgradeable = isValid && versionForMigration.state === VersionState.upgradeable;
+  if (isUpgradeable) {
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:186]: misses `checkMethod(ctx)` in original if condition

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 2]
```diff
@@ -923,6 +923,20 @@ export async function generateApimPluginEnvContent(context: MigrationContext): P
...
+  try {
+    const projectSettings = await loadProjectSettings(projectPath);
+    const solutionSettings = projectSettings.solutionSettings;
+    if (projectSettings && solutionSettings && solutionSettings.activeResourcePlugins) {
+      return true;
+    } else {
+      return false;
+    }
+  } catch (error) {
+    return false;
```
Comment (blackchoey) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:936]: should we add some telemetry about why it fails?

Comment (blackchoey) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts:936]: Maybe also add some telemetry to tell how many projects have invalid projectSettings.json

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts (hunk 1 above)
Comment (xzf0587) [reply packages/fx-core/src/core/middleware/projectMigratorV3.ts:186]: CheckMethod condition is correct in v5. 
It excludes  getProjectConfig and checkPermission which are needed to add upgrade logic. So remove the condition.

**Review by blackchoey — APPROVED** [review]

**Comment by blackchoey** [comment]
Please add some test cases for the change

**Review by jayzhang — APPROVED** [review]

**Pull request merged**